core/node/libp2p.go

```diff
@@ -0,0 +1,587 @@
+}
+
+type PNetFingerprint []byte
+
+func P2PPNet(repo repo.Repo) (opts Libp2pOpts, fp PNetFingerprint, err error) {
+	swarmkey, err := repo.SwarmKey()
+	if err != nil || swarmkey == nil {
+		return opts, nil, err
+	}
+
+	protec, err := pnet.NewProtector(bytes.NewReader(swarmkey))
+	if err != nil {
+		return opts, nil, fmt.Errorf("failed to configure private network: %s", err)
+	}
+	fp = protec.Fingerprint()
+
+	opts.Opts = append(opts.Opts, libp2p.PrivateNetwork(protec))
+	return opts, fp, nil
+}
+
+func P2PPNetChecker(repo repo.Repo, ph host.Host, lc fx.Lifecycle) error {
+	// TODO: better check?
+	swarmkey, err := repo.SwarmKey()
+	if err != nil || swarmkey == nil {
+		return err
+	}
+
+	done := make(chan struct{})
+	lc.Append(fx.Hook{
+		OnStart: func(_ context.Context) error {
+			go func() {
+				t := time.NewTicker(30 * time.Second)
+				defer t.Stop()
+
+				<-t.C // swallow one tick
+				for {
+					select {
+					case <-t.C:
+						if len(ph.Network().Peers()) == 0 {
+							log.Warning("We are in private network and have no peers.")
+							log.Warning("This might be configuration mistake.")
+						}
+					case <-done:
+						return
+					}
+				}
+			}()
+			return nil
+		},
+		OnStop: func(_ context.Context) error {
+			close(done)
+			return nil
+		},
+	})
+	return nil
+}
+
+func makeAddrsFactory(cfg config.Addresses) (p2pbhost.AddrsFactory, error) {
+	var annAddrs []ma.Multiaddr
+	for _, addr := range cfg.Announce {
+		maddr, err := ma.NewMultiaddr(addr)
+		if err != nil {
+			return nil, err
+		}
+		annAddrs = append(annAddrs, maddr)
+	}
+
+	filters := mafilter.NewFilters()
+	noAnnAddrs := map[string]bool{}
+	for _, addr := range cfg.NoAnnounce {
+		f, err := mamask.NewMask(addr)
+		if err == nil {
+			filters.AddDialFilter(f)
+			continue
+		}
+		maddr, err := ma.NewMultiaddr(addr)
+		if err != nil {
+			return nil, err
+		}
+		noAnnAddrs[maddr.String()] = true
+	}
+
+	return func(allAddrs []ma.Multiaddr) []ma.Multiaddr {
+		var addrs []ma.Multiaddr
+		if len(annAddrs) > 0 {
+			addrs = annAddrs
+		} else {
+			addrs = allAddrs
+		}
+
+		var out []ma.Multiaddr
+		for _, maddr := range addrs {
+			// check for exact matches
+			ok := noAnnAddrs[maddr.String()]
+			// check for /ipcidr matches
+			if !ok && !filters.AddrBlocked(maddr) {
+				out = append(out, maddr)
+			}
+		}
+		return out
+	}, nil
+}
+
+func P2PAddrsFactory(cfg *config.Config) (opts Libp2pOpts, err error) {
+	addrsFactory, err := makeAddrsFactory(cfg.Addresses)
+	if err != nil {
+		return opts, err
+	}
+	if !cfg.Swarm.DisableRelay {
+		addrsFactory = composeAddrsFactory(addrsFactory, filterRelayAddrs)
+	}
+	opts.Opts = append(opts.Opts, libp2p.AddrsFactory(addrsFactory))
+	return
+}
+
+func filterRelayAddrs(addrs []ma.Multiaddr) []ma.Multiaddr {
+	var raddrs []ma.Multiaddr
+	for _, addr := range addrs {
+		_, err := addr.ValueForProtocol(circuit.P_CIRCUIT)
+		if err == nil {
+			continue
+		}
+		raddrs = append(raddrs, addr)
+	}
+	return raddrs
+}
```

Comment:
This should no longer be necessary as of the latest master. Instead of introducing merge conflicts, can we just remove this here?

Reply:
done